src/Browscap/Coverage/Processor.php

```diff
@@ -0,0 +1,571 @@
+        'JSONString' => 3,
+        'STRING' => 4,
+        'JSONNumber' => 5,
+        'NUMBER' => 6,
+        'JSONNullLiteral' => 7,
+        'NULL' => 8,
+        'JSONBooleanLiteral' => 9,
+        'TRUE' => 10,
+        'FALSE' => 11,
+        'JSONText' => 12,
+        'JSONValue' => 13,
+        'EOF' => 14,
+        'JSONObject' => 15,
+        'JSONArray' => 16,
+        '{' => 17,
+        '}' => 18,
+        'JSONMemberList' => 19,
+        'JSONMember' => 20,
+        ':' => 21,
+        ',' => 22,
+        '[' => 23,
+        ']' => 24,
+        'JSONElementList' => 25,
+        '$accept' => 0,
+        '$end' => 1,
+    ];
+
+    /**
+     * @var array
+     */
+    private $symbolLookup = [];
+
+    /**
+     * @var array
+     */
+    private $coveredIds = [];
+
+    /**
+     * @var array
+     */
+    private $coverage = [];
+
+    /**
+     * @var int
+     */
+    private $funcCount = 0;
+
+    /**
+     * Create a new Coverage Processor for the specified directory
+     *
+     * @param string $resourceDir
+     */
+    public function __construct($resourceDir)
+    {
+        $this->resourceDir = $resourceDir;
+
+        $this->symbolLookup = array_flip($this->symbols);
+    }
+
+    /**
+     * Process the directory of JSON files using the collected pattern ids
+     *
+     * @param array $coveredIds
+     */
+    public function process(array $coveredIds)
+    {
+        $this->setCoveredPatternIds($coveredIds);
+
+        $iterator = new \RecursiveDirectoryIterator($this->resourceDir);
+
+        foreach (new \RecursiveIteratorIterator($iterator) as $file) {
+            /** @var $file \SplFileInfo */
+            if (!$file->isFile() || $file->getExtension() !== 'json') {
+                continue;
+            }
+
+            $patternFileName = substr($file->getPathname(), strpos($file->getPathname(), 'resources/'));
+
+            if (!isset($this->coveredIds[$patternFileName])) {
+                $this->coveredIds[$patternFileName] = [];
+            }
+
+            $this->coverage[$patternFileName] = $this->processFile(
+                $patternFileName,
+                file_get_contents($file->getPathname()),
+                $this->coveredIds[$patternFileName]
+            );
+        }
+    }
+
+    /**
+     * Write the coverage data in JSON format to specified filename
+     *
+     * @param string $fileName
+     */
+    public function write($fileName)
+    {
+        file_put_contents(
+            $fileName,
+            // str_replace here is to convert empty arrays into empty JS objects, which is expected by
+            // codecov.io. Owner of the service said he was going to patch it, haven't tested since
+            // Note: Can't use JSON_FORCE_OBJECT here as we **do** want arrays for the 'b' structure
+            // which FORECE_OBJECT turns into objects, breaking at least the Istanbul coverage reporter
+            str_replace('[]', '{}', json_encode($this->coverage, JSON_UNESCAPED_SLASHES))
```

Comment:
As mentioned in the comment there, no. The reason being, there are array structures in the coverage.json file that need to remain as arrays.  In the "b" structure (which is where the coverage data for different branches are stored) each entry is an array showing which branches of the branch statement are covered.  Which basically looks like this:

```json
"b": {
  "1": [0,1,2],
  "2": [0,0]
}
```

If I use the JSON_FORCE_OBJECT option, that turns into something like this:

```json
"b": {
  "1": {"0": 0, "1": 1, "2": 2},
  "2": {"0": 0, "1": 0}
}
```

Which makes the Istanbul reporter throw an error:

>~/.npm_packages/lib/node_modules/istanbul/lib/report/lcovonly.js:79
>            branchArray.forEach(function (b) {
>                        ^
>
>TypeError: branchArray.forEach is not a function

`branchArray` in this case is those sections in the "b" hash that should be arrays.